pytest_splunk_addon_ui_smartx/components/controls/checkboxgroup.py

```diff
@@ -0,0 +1,137 @@
+
+        Args:
+            group_name (str): The name of the checkbox group to check.
+
+        Returns:
+            bool: True if the group is expanded, False otherwise.
+        """
+        self.elements.update(
+            {
+                f"{group_name}_button": Selector(
+                    by=By.XPATH,
+                    select=self.elements.get("container").select
+                    + f'//span[text()="{group_name}"]/ancestor::button',
+                )
+            }
+        )
+        return (
+            getattr(self, f"{group_name}_button").get_attribute("aria-expanded")
+            == "true"
+        )
+
+    def get_checkbox(self, checkbox_name: str) -> Checkbox:
+        return Checkbox(
+            self.browser,
+            Selector(
+                by=By.XPATH,
+                select=self.elements.get("container").select
+                + f"//div[@data-test-field='{checkbox_name}']/parent::div",
+            ),
+        )
+
+    def select_checkbox_and_set_value(
+        self, group_name: str, checkbox_name: str, checkbox_value: str = None
+    ) -> None:
+        """
+        Expands a group and selects a checkbox, then sets the specified value.
+
+        Args:
+            group_name (str): The name of the group to expand.
+            checkbox_name (str): The name of the checkbox to select.
+            checkbox_value (str): The value to set for the checkbox.
+        """
+        self.expand_group(group_name)
+        self.get_checkbox(checkbox_name).check()
+        if checkbox_value:
+            self.get_textbox(checkbox_name).set_value(checkbox_value)
+
+    def deselect(self, group_name: str, checkbox_name: str) -> None:
+        """
+        Expands a group and deselects the specified checkbox.
+
+        Args:
+            group_name (str): The name of the group to expand.
+            checkbox_name (str): The name of the checkbox to deselect.
+        """
+        self.expand_group(group_name)
+        self.get_checkbox(checkbox_name).uncheck()
+
+    def get_textbox(self, checkbox_name: str) -> TextBox:
+        return TextBox(
+            self.browser,
+            Selector(
+                by=By.XPATH,
+                select=self.elements.get("container").select
+                + f"//div[@data-test-field='{checkbox_name}' and @data-test='number']",
+            ),
+        )
+
+    def get_checkbox_text_value(self, group_name: str, checkbox_name: str) -> str:
+        """
+        Expands a group and retrieves the text value of the specified checkbox.
+
+        Args:
+            group_name (str): The name of the group to expand.
+            checkbox_name (str): The name of the checkbox to retrieve the value from.
+
+        Returns:
+            str: The value of the checkbox.
+        """
+        self.expand_group(group_name)
+        return self.get_textbox(checkbox_name).get_value()
+
+    def expand_group(self, group_name: str) -> None:
```

Comment:
Does it make sense to add method collapse_group?

Reply:
We can add but it as a functionality, but it won't block other browser interactions.
For eg: if a dropdown is open clicking other components may result in stale element or element not intractable error but not in this case.

Reply:
Yes I do agree. But I consider expanding/collapsing as a part of the functionality which may/may not work. So how would we confirm, if pressing the collapse button, actually collapsed group section without such method here?
Or maybe it is covered in other scenario?

Reply:
Yes so expanding and those scenarios are covered in main functionality like select_checkbox_and_set_value.
For component level testing we can add that in UCC